boards/seeedstudio-gd32/include/periph_conf.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2020 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_seeedstudio-gd32
+ * @{
+ *
+ * @file
+ * @brief       Board specific definitions for the SeeedStudio GD32 RISC-V board
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef PERIPH_CONF_H
+#define PERIPH_CONF_H
+
+#include "macros/units.h"
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* This board provides an high frequency oscillator */
+#ifndef CONFIG_BOARD_HAS_HXTAL
+#define CONFIG_BOARD_HAS_HXTAL        1
+#endif
+
+#define CLOCK_HXTAL         MHZ(8)             /**< HXTAL frequency */
+#define CLOCK_CORECLOCK     MHZ(104)           /**< CPU clock frequency in Hz */
```

Comment:
@bergzand Why did you define 104 MHz as `CLOCK_CORECLOCK` and not 108 MHz?

Reply:
Tough question, can't say I remember the reason after leaving this alone for a year.

Reply:
Might it be that the resulting UART baud rate was wrong and you had to adjust it a bit :wink: While enabling `periph_usbdev` for GD32VF103, I realized that the UART baud rate was 107.066 baud instead of 115.200 so that the I couldn't connect. The resulting USB speed was 11,11 Mbps instead of 12 Mbps. When investigating the problem, I figured out that `cpu/gd32v/clock.c` is buggy.

One problem is that `CLOCK_HXTAL` here is a value and not a flag, so shifting to the left changes anything in the register but does not set the `PLLSEL` bit. https://github.com/RIOT-OS/RIOT/blob/d193988db273803e614f95bc28d1564ff8eb8e90/cpu/gd32v/clock.c#L131-L133 Thus not the predivided HXTAL is used as PLL input but the half IRC8M clock. The second problem following as consequence of this problem is that setting the `RCU_CTL` register to the IRC8M bit also removes the IRC8M calibration and trim adjust value in this register :worried: https://github.com/RIOT-OS/RIOT/blob/d193988db273803e614f95bc28d1564ff8eb8e90/cpu/gd32v/clock.c#L122 As consequence the quite incorrect IRC8M clock is used. I will provide a fix with which the clock clock settings, the UART and the USB peripherals work, with and without HXTAL :smile: